src/main/java/pl/javastart/exercise/streams/FilteringMachine.java

```diff
@@ -1,23 +1,36 @@
 package pl.javastart.exercise.streams;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class FilteringMachine {
 
     public List<Integer> filterOutNotEvenNumbers(List<Integer> numberList) {
-        return null;
+        List<Integer> collectionToFilter = numberList.stream()
```

Comment:
Tak, lepiej jest od razu do returna wrzucać wynik jeżeli wiesz że kod będzie jednolinjkowcjem.

Reply:
Jakby nie patrzeć deklaracja zmiennej to nic innego jak referencja który zajmuje (w zależności od wersji Javy x32 / x64) 4 lub 8 bajtów. Więc można AŻ tyle zaoszczędzić :)

Reply:
Tak swoją drogą w Javie x64 mamy też kompresowane wskaźniki które zajmują też 4 bajty. Dodaje to ponieważ męczy mnie to że nie do końca pradę tutaj napisałem.